pkg/trace/api/otlp.go

```diff
@@ -363,39 +357,85 @@ func convertSpan(rattr map[string]string, lib pdata.InstrumentationLibrary, in p
 	if in.Events().Len() > 0 {
 		span.Meta["events"] = marshalEvents(in.Events())
 	}
+	var ctags strings.Builder // collect container tags from attributes
 	in.Attributes().Range(func(k string, v pdata.AttributeValue) bool {
 		switch v.Type() {
 		case pdata.AttributeValueTypeDouble:
 			span.Metrics[k] = v.DoubleVal()
 		case pdata.AttributeValueTypeInt:
 			span.Metrics[k] = float64(v.IntVal())
 		default:
-			span.Meta[k] = v.AsString()
+			switch k {
+			case "operation.name":
+				span.Name = v.AsString()
+			case "service.name":
```

Comment:
Whenever possible, use the constants from https://pkg.go.dev/go.opentelemetry.io/collector/model@v0.48.0/semconv/v1.6.1 instead; at some point we need to add support for something called 'telemetry schemas' and it will be easier if we use the `semconv` package.

Reply:
Only the name is a coincidence. These string literals are actually from the Go tracer, where we also allow overriding these span fields by means of tag (attribute) keys. They are available there as public constants, but didn't want to import that module just for that... Do you think I should?

Reply:
> These string literals are actually from the Go tracer, where we also allow overriding these span fields by means of tag (attribute) keys.

Not sure I get this 🤔 Do you mean the OpenTelemetry Collector Datadog traces exporter? Or does our dd-trace-go Go tracer allow exporting in OTLP format?

> They are available there as public constants, but didn't want to import that module just for that...

We already import it https://github.com/DataDog/datadog-agent/blob/62f28598d2cbe8dccf5a9c56260fbc264b256d33/pkg/trace/api/otlp.go#L33 so in that sense we are fine

Reply:
What I mean is that the Datadog span has fields called `Name`, `Service`, `Resource` and `Type`. They are unsettable from an OpenTelemetry SDK so we provide a shortcut for setting these by allowing users to manually set `operation.name`, `service.name`, `resource.name` or `span.type` attributes/tags, which automatically translate to those fields.

We have been using these since before OpenTelemetry for the same reason in OpenTracing back in the days.

See:
* [Usage in tracer](https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/span.go#L267-L278)
* [Their definition](https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/ext/tags.go#L36-L51)

These shortcuts are unrelated to OpenTelemetry, it's just a coincidence that some of them have the same name. I hope it makes more sense now.

Reply:
Ah, that makes more sense, thanks for the link. Yeah then this suggestion doesn't make sense, let me resolve it!